cmd/kube-controller-manager/app/core.go

```diff
@@ -156,13 +156,13 @@ func startNodeIpamController(ctx ControllerContext) (http.Handler, bool, error)
 	}
 
 	var nodeCIDRMaskSizeIPv4, nodeCIDRMaskSizeIPv6 int
-	if utilfeature.DefaultFeatureGate.Enabled(features.IPv6DualStack) {
+	if dualStack {
```

Comment:
`dualStack` will be `true` when the `--cluster-cidr` is configured as dual-stack (at least one from each IPFamily) regardless of the `IPv6DualStack` feature gate is enabled or not. So even `IPv6DualStack` feature gate is enabled by default, if the user does not set `--cluster-cidr` as dual-stack, we will not enter this `if` branch, `--node-cidr-mask-size` is valid as before.